File: java/src/tugas_pbo/tugas3/Main.dart
import 'BangunDatar.dart';
import 'Persegi.dart';
import 'PersegiPanjang.dart';
import 'Segitiga.dart';

void main() {
  // Menggunakan satu tipe variabel (BangunDatar) untuk berbagai objek
  List<BangunDatar> daftarBangun = [
    BangunDatar(3.0, 4.0),
    Persegi(5.0),
    PersegiPanjang(6.0, 2.0),
    Segitiga(4.0, 5.0)
  ];

  // Menampilkan semua isi list
  for (var bangun in daftarBangun) {
    bangun.tampilkan();
    print(""); // Memberi jarak antar output
  }
}

File: java/src/tugas_pbo/tugas3/Main.java
package tugas_pbo.tugas3;

public class Main {
    public static void main(String[] args) {
        // Menggunakan satu tipe variabel yang sama (Polymorphism)
        BangunDatar bd;

        // Cetak Bangun Datar Dasar
        bd = new BangunDatar(3.0, 4.0);
        bd.tampilkan();
        System.out.println();

        // Cetak Persegi
        bd = new Persegi(5.0);
        bd.tampilkan();
        System.out.println();

        // Cetak Persegi Panjang
        bd = new PersegiPanjang(6.0, 4.0);
        bd.tampilkan();
        System.out.println();

        // Cetak Segitiga
        bd = new Segitiga(10.0, 5.0);
        bd.tampilkan();
    }
}